src/clj/fluree/db/flake/reasoner.cljc

```diff
@@ -163,28 +165,47 @@
     (log/debug "Reasoner - source OWL rules: " graph)
     (owl-datalog/owl->datalog inserts graph)))
 
+(defn rules-from-dbs
+  [methods inserts dbs]
+  (go-try
```

Comment:
This function has way too much nesting. You should break it up into multiple functions that compute the steps you need and put them together here.